### PR TITLE
Catalog edit link

### PR DESCRIPTION
**Pull request by hooloovooo**
## Hey, I just made a Pull Request!

Adds the edit link to catalog list.

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

2020-06-05-110346_710x331_scrot

### CODE REVIEW

**Review by stefanalund — APPROVED** [review]

File: plugins/catalog/src/data/component.ts [hunk 1]
```diff
@@ -13,11 +13,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import React from 'react';
```
Comment (shmidt-i) [inline plugins/catalog/src/data/component.ts:16]:
```suggestion
import { ReactNode } from 'react';
```

File: plugins/catalog/src/data/component.ts [hunk 2]
```diff
@@ -13,11 +13,12 @@
...
 import { Location } from '@backstage/catalog-model';
 
 export type Component = {
   name: string;
   kind: string;
-  description: string;
+  description: React.ReactNode;
```
Comment (shmidt-i) [inline plugins/catalog/src/data/component.ts:22]:
```suggestion
  description: ReactNode;
```

File: plugins/catalog/src/data/utils.tsx [hunk 3]
```diff
@@ -13,8 +13,19 @@
...
+import Edit from '@material-ui/icons/Edit';
+import IconButton from '@material-ui/core/IconButton';
+import { styled } from '@material-ui/core/styles';
+
+const DescriptionWrapper = styled('span')({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const createEditLink = (url: string): string => url.replace('blob', 'edit');
```
Comment (shmidt-i) [inline plugins/catalog/src/data/utils.tsx:28]: tricky if username or repo have 'blob' as a part of the string :)

Comment (hooloovooo) [reply plugins/catalog/src/data/utils.tsx:28]: True, this is veeeeery shaky. Creating a new PR with a slightly improved one.

File: plugins/catalog/src/data/utils.tsx [hunk 4]
```diff
@@ -23,7 +34,18 @@ export function envelopeToComponent(
   return {
     name: envelope.metadata?.name ?? '',
```
Comment (freben) [inline plugins/catalog/src/data/utils.tsx:35]: both metadata and name are always present (so you can also remove the question mark on all metadata below)

File: plugins/catalog/src/data/utils.tsx [hunk 5]
```diff
@@ -23,7 +34,18 @@ export function envelopeToComponent(
...
     kind: envelope.kind ?? 'unknown',
```
Comment (freben) [inline plugins/catalog/src/data/utils.tsx:36]: kind is also always present

**Pull request merged**